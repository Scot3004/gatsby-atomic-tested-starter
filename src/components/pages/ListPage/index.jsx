import React from 'react';
import PropTypes from 'prop-types';

import Layout from '../../layouts';
import PostList from '../../organisms/PostList';

export default function ListPage({ data }) {
  return (
    <Layout title={data.site.siteMetadata.title}>
      <PostList data={data.allMarkdownRemark.edges} />
    </Layout>
  );
}

ListPage.propTypes = {
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        title: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  }).isRequired
};
