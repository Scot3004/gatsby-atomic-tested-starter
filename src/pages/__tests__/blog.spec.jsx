import React from 'react';
import 'jest-styled-components';
import { graphql } from 'gatsby';
import renderer from 'react-test-renderer';
import ListPage from '../blog';

const data = {
  site: {
    siteMetadata: {
      title: 'Molecular starter',
      menuLinks: []
    }
  },
  allMarkdownRemark: {
    edges: [
      {
        node: {
          id: '39cb6d6b-1bf3-5aa1-a310-b882314db77b',
          excerpt:
            '"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi utâ€¦',
          frontmatter: {
            date: 'November 07, 2017',
            title: 'My first blog post',
            image: 'https://www.secorto.com/img/logo.png'
          },
          fields: {
            slug: '/blog/my-first-post'
          }
        }
      }
    ]
  }
};
test('query correctly', () => {
  const tree = renderer.create(<ListPage data={data} />).toJSON();
  expect(graphql).toBeCalled();
  expect(tree).toMatchSnapshot();
});
