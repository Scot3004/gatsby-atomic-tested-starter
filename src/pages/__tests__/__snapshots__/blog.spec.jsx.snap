// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: minmax(auto,64em);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
}

.c3 {
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  width: 64px;
  height: 64px;
  padding: 1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 1rem 0;
  padding: .6rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
}

.c2:hover {
  box-shadow: 0 6px 12px 0 rgba(0,0,0,0.2);
}

.c2:last-child {
  margin: 0;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 {
  margin: 0;
}

.c5 {
  margin: 0 0 .6rem 0;
  padding: 0;
  word-wrap: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <header>
      <h1>
        Molecular starter
      </h1>
    </header>
    <div
      className="c2"
    >
      <img
        alt="My first blog post"
        className="c3"
        size="64px"
        src="https://www.secorto.com/img/logo.png"
      />
      <div
        className="c4"
      >
        <a
          href="/blog/my-first-post"
        >
          <h2
            className="c5"
          >
            My first blog post
          </h2>
        </a>
        <p
          className="c6"
        >
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi utâ€¦
        </p>
      </div>
    </div>
  </div>
</div>
`;
